### PR TITLE
Add support for SearchApi API

### PR DESCRIPTION
**Pull request by SebastjanPrachovskij**
We have integrated a new webSearch - [SearchApi](https://www.searchapi.io/), following a similar format to the other SERP providers.

- Feel free to assign @SebastjanPrachovskij as a main reviewer for any SearchApi-related searches. We will be glad to help and support chat-ui development.

### CODE REVIEW

File: src/lib/server/websearch/search/endpoints/searchApi.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { env } from "$env/dynamic/private";
+import type { WebSearchSource } from "$lib/types/WebSearch";
+import { isURL } from "$lib/utils/isUrl";
```
Comment (Saghen) [inline src/lib/server/websearch/search/endpoints/searchApi.ts:3]: `'isURL' is defined but never used`

**Review by Saghen — COMMENTED** [review]
thanks for the pr! couple small changes and a rebase/merge needed

File: README.md [hunk 2]
```diff
@@ -166,7 +166,7 @@ PUBLIC_APP_DISCLAIMER=
 
 ### Web Search config
 
-You can enable the web search through an API by adding `YDC_API_KEY` ([docs.you.com](https://docs.you.com)) or `SERPER_API_KEY` ([serper.dev](https://serper.dev/)) or `SERPAPI_KEY` ([serpapi.com](https://serpapi.com/)) or `SERPSTACK_API_KEY` ([serpstack.com](https://serpstack.com/)) to your `.env.local`.
+You can enable the web search through an API by adding `YDC_API_KEY` ([docs.you.com](https://docs.you.com)) or `SERPER_API_KEY` ([serper.dev](https://serper.dev/)) or `SERPAPI_KEY` ([serpapi.com](https://serpapi.com/)) or `SERPSTACK_API_KEY` ([serpstack.com](https://serpstack.com/)) or `SEARCHAPI_KEY` ([searchapi.io](https://www.searchapi.io/)) to your `.env.local`.
```
[the remark this answers is not in the document]
Comment (SebastjanPrachovskij) [reply README.md:184]: Hey Saghen, 

Thanks for the review!

Should I still remove it? I saw #1207 PR which returns a readme file.

**Review by nsarrazin — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**